From 35cedac3faa5263a19ba8c7a187f3649436e0626 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Fri, 19 Oct 2018 19:22:54 +0300
Subject: [PATCH 86/86] ucm-imx8: touch: Add an option for another pendown
 interrupt

Add an option for another pendown interrupt.
Revisoon 1.1 will have it.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/sb-ucm-imx8.dtsi      | 5 +++++
 arch/arm64/boot/dts/compulab/sbc-ucm-imx8-lvds.dts | 9 ++++++---
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/compulab/sb-ucm-imx8.dtsi b/arch/arm64/boot/dts/compulab/sb-ucm-imx8.dtsi
index e70cc2b0..560b787 100644
--- a/arch/arm64/boot/dts/compulab/sb-ucm-imx8.dtsi
+++ b/arch/arm64/boot/dts/compulab/sb-ucm-imx8.dtsi
@@ -19,6 +19,11 @@
 		        MX8MQ_IOMUXC_GPIO1_IO15_GPIO1_IO15 0x16
 			>;
 		};
+		touchscreen_pins: touchscreen-pins {
+			fsl,pins = <
+				MX8MQ_IOMUXC_UART2_TXD_GPIO5_IO25 0x17059
+			>;
+		};
     };
 };
 
diff --git a/arch/arm64/boot/dts/compulab/sbc-ucm-imx8-lvds.dts b/arch/arm64/boot/dts/compulab/sbc-ucm-imx8-lvds.dts
index 00b3748..293d48a 100644
--- a/arch/arm64/boot/dts/compulab/sbc-ucm-imx8-lvds.dts
+++ b/arch/arm64/boot/dts/compulab/sbc-ucm-imx8-lvds.dts
@@ -31,10 +31,13 @@
         compatible = "goodix,gt911";
         reg = <0x5d>;
 
-        interrupt-parent = <&pca9555>;
-        interrupts = <13 IRQ_TYPE_EDGE_FALLING>;
+        pinctrl-names = "default";
+        pinctrl-0 = <&touchscreen_pins>;
 
-        irq-gpios = <&pca9555 13 GPIO_ACTIVE_HIGH>;
+        interrupt-parent = <&gpio5>;
+        interrupts = <25 IRQ_TYPE_EDGE_FALLING>;
+
+        irq-gpios = <&gpio5 25 GPIO_ACTIVE_HIGH>;
         reset-gpios = <&pca9555 14 GPIO_ACTIVE_HIGH>;
         status = "okay";
     };
-- 
1.9.1

